plugin.rb

```diff
@@ -69,5 +70,79 @@ class Engine < ::Rails::Engine
     add_to_serializer(:basic_category, key.to_sym) { object.custom_fields[key] }
   end
 
+  class ::TopicPreviews::ThumbnailselectionController < ::ApplicationController
```

Comment:
@merefield This needs to go in a "controllers" folder and imported into the plugin.rb

Reply:
Ah yes, no problem, better than bloating plugin.rb

Reply:
The other solution to the bug you rightly highlight is to try and control the visibility of the button and hide it before the Topic is created for the first time.  I wanted to keep the interface a modal as it's nice and neat this way ... creating a separate route will make this chooser appear in its own path and to me that's not very elegant?